### DISCUSSION TITLE
Write values to multiple nodes at the same time

### DISCUSSION THREAD

**Discussion by vidyarao123** [FreeOpcUa/python-opcua#1280] [Q&A]
I have a list of 10 nodeIDs. Usually to update a value we use the .set_value() method. General way of achieving this would be to update them using a for loop and iterate over the list. 
I have another list which has 10 integer values. But instead of going through a cyclic way(for loop) I want to update all 10 nodeIDs at the same time from the second list which has it's corresponding values. Is there a way to achieve this ? Is it possible to write values to multiple node IDs at the same time. ?

Thanks

**Comment by AndreasHeine** [comment]
are you server or client?

**Reply by vidyarao123** [reply]
Server

**Reply by AndreasHeine** [reply]
the server does not have a method like this....

instead of node.set_value() you could use:
server.iserver.set_attribute_value(nodeid, datavalue, attr=ua.AttributeIds.Value)
but still you must loop!

https://github.com/FreeOpcUa/python-opcua/blob/67e5484c5fc05aaf4b885588a767251605b79a4b/opcua/server/internal_server.py#L279

**Reply by AndreasHeine** [reply]
if you have two lists just list(zip(nodeids, values)) together and then loop over it and set the value (set_attribute_value(nodeid, datavalue, attr=ua.AttributeIds.Value))

**Reply by vidyarao123** [reply]
Hi, I have almost of 30000 tags. So looping over from 1 to 30000 will be a time consuming task. And since I am getting the data from Modbus, iterating one after that will be a tedious task that's why I was wondering if there is a way to write all 30000 tags at once Thanks

**Reply by AndreasHeine** [reply]
there is no other way on server side...

clients can write multiple nodes with one request but internally we loop over them!

**Reply by oroulet** [reply]
the fastest methode on server side is write_attrbute (opcua-asyncio only). It write directly to the target dict. But with 30000 nodes you might be over what Python can do
